### ISSUE TITLE
'make test' fails at 0.7.0 version

### ISSUE THREAD

**Issue by anatol** [crystal-lang/shards#140]
If I checkout 0.7.0 and try to build/test it fails

```
  crystal build --release -o bin/shards src/shards.cr
  ./bin/shards install
  make test
```

```
Updating https://github.com/ysbaddaden/minitest.cr.git
Installing minitest (0.3.4)
make test_unit
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
make[1]: Entering directory '/home/anatol/sources/community-arch/shards/trunk/src/shards-0.7.0'
/usr/bin/crystal run test/*_test.cr -- --parallel=1
Error in ./test/git_resolver_test.cr:1: while requiring "./test_helper"

require "./test_helper"
^

in ./test/test_helper.cr:1: while requiring "minitest/autorun": can't find file 'minitest/autorun' relative to '/home/anatol/sources/community-arch/shards/trunk/src/shards-0.7.0/test'

require "minitest/autorun"
^

make[1]: *** [Makefile:55: test_unit] Error 1
make[1]: Leaving directory '/home/anatol/sources/community-arch/shards/trunk/src/shards-0.7.0'
make: *** [Makefile:50: test] Error 2
```

**Comment by RX14** [comment]
What version of crystal are you using when running the specs?

**Comment by asterite** [comment]
You also need to run `shards` before running `make test`. I just tried it but I had to use the master version of `minitest`, otherwise it would give me a syntax error

**Comment by anatol** [comment]
Tested it on Arch Linux with both `Crystal 0.19.4 (2016-10-07)` and `Crystal 0.20.0 (2016-11-22)`

**Comment by anatol** [comment]
@asterite I am already running `./bin/shards install` right before `make test`. Is there any additional `shard` command is needed?

**Comment by RX14** [comment]
I can reproduce this on shards `0.7.0` and crystal `0.20.0`

**Comment by ysbaddaden** [comment]
I'm afraid I can't reproduce the issue (ubuntu 14.04, official crystal 0.20.0 deb package).

Is it possible that the Archlinux package has a wrong `CRYSTAL_PATH`? one that mentions `libs` when it should now mention `lib`? as now required by both Shards 0.7 and Crystal 0.20. See #127 for details.

**Comment by anatol** [comment]
Arch crystal build script is here https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/crystal

we compile crystal as
```
make release=1 \
       PATH="$srcdir/$pkgname-$pkgver-$_binary_rel/bin:$PATH" \
       CRYSTAL_PATH="$srcdir/$pkgname-$pkgver/src" \
       CRYSTAL_CONFIG_VERSION="$pkgver" \
       CRYSTAL_CONFIG_PATH="libs:/usr/lib/crystal" \
       CRYSTAL_CACHE_DIR="/tmp/crystal"
  make doc CRYSTAL_CACHE_DIR="/tmp/crystal"
```

**Comment by ysbaddaden** [comment]
`CRYSTAL_CONFIG_PATH` is wrong: it states `libs` when it should have been `lib:libs` in 0.19 and should now be `lib` in 0.20.

**Comment by anatol** [comment]
Just to clarify my `crystal` build script should change the line `CRYSTAL_CONFIG_PATH="libs:/usr/lib/crystal"` to `CRYSTAL_CONFIG_PATH="lib"`. Anything else need to changed?

**Comment by RX14** [comment]
@anatol should be `CRYSTAL_CONFIG_PATH="lib:/usr/lib/crystal"`

**Comment by anatol** [comment]
+ @jhass

I modified crystal's `CRYSTAL_CONFIG_PATH` to `"lib:/usr/lib/crystal"` but shards `make test` still fails, this time with following error:

```
==> Starting build()...
Warning: MemoryIO is deprecated and will be removed after 0.20.0, use IO::Memory instead
Warning: MemoryIO is deprecated and will be removed after 0.20.0, use IO::Memory instead
Warning: MemoryIO is deprecated and will be removed after 0.20.0, use IO::Memory instead
==> Starting check()...
Updating https://github.com/ysbaddaden/minitest.cr.git
Installing minitest (0.3.4)
make test_unit
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
make[1]: Entering directory '/build/shards/src/shards-0.7.0'
/usr/bin/crystal run test/*_test.cr -- --parallel=1
Error in test/git_resolver_test.cr:1: while requiring "./test_helper"

require "./test_helper"
^

in test/test_helper.cr:1: while requiring "minitest/autorun"

require "minitest/autorun"
^

in lib/minitest/src/autorun.cr:1: while requiring "./minitest"

require "./minitest"
^

in lib/minitest/src/minitest.cr:6: while requiring "./minitest/spec"

require "./minitest/spec"
^

in lib/minitest/src/minitest/spec.cr:1: while requiring "./expectations"

require "./expectations"
^

Syntax error in lib/minitest/src/minitest/expectations.cr:163: expecting token 'EOF', not 'end'

end
^

make[1]: *** [Makefile:55: test_unit] Error 1
make[1]: Leaving directory '/build/shards/src/shards-0.7.0'
make: *** [Makefile:50: test] Error 2
==> ERROR: A failure occurred in check().
    Aborting...
==> ERROR: Build failed, check /var/lib/archbuild/extra-x86_64/anatol/build
```

**Comment by Sija** [comment]
@anatol `ysbaddaden/minitest.cr` needs to at least at `v0.3.5` which fixes Crystal 0.20 compatibility.

**Comment by ysbaddaden** [comment]
Sounds like there is an issue with v0.7.0. Does it reproduce on master?

**Comment by anatol** [comment]
shards from `HEAD` (commit `4fd5f8fcb43`) compiles and passes test at my Arch Linux with Crystal `0.20.0`. Thanks! Could you please make another shards release with the recent fixes?

**Comment by Sija** [comment]
`minitest` dependency from https://github.com/crystal-lang/shards/blob/v0.7.0/shard.yml should dl latest version, that is `v0.3.5`, which is compatible with Crystal 0.20 so it should work?

**Comment by anatol** [comment]
@Sija the version in that release is locked to `0.3.4` https://github.com/crystal-lang/shards/blob/v0.7.0/shard.lock

HEAD has changed it to `0.3.5`.

**Comment by Sija** [comment]
@anatol You're right, thanks for pointing that out.

**Comment by ysbaddaden** [comment]
I just released v0.7.1
